metadata/leases.go

```diff
@@ -461,6 +475,59 @@ func removeIngestLease(ctx context.Context, tx *bolt.Tx, ref string) error {
 	return bkt.Delete([]byte(ref))
 }
 
+func addImageLease(ctx context.Context, tx *bolt.Tx, ref string, labels map[string]string) error {
+	lid, ok := leases.FromContext(ctx)
+	if !ok {
+		return nil
+	}
+
+	// If image doesn't have expiration, it does not need to be leased
+	if _, ok := labels[string(labelGCExpire)]; !ok {
```

Comment:
Suggest to add integration test case for this. I think it will be a good example for developer.

Reply:
Let me look into adding a test under images for this, we don't usually unit test the non exported functions. I did find another fix for one of the tests though